internal/controller/driver_controller.go

```diff
@@ -741,7 +823,26 @@ func (r *driverReconcile) reconcileControllerPluginDeployment() error {
 								),
 							})
 						}
-
+						// CSI Logrotate Container
+						if logRotator != nil {
+							resources := ptr.Deref(pluginSpec.Resources.LogRotator, corev1.ResourceRequirements{})
+							containers = append(containers, corev1.Container{
+								Name:            "log-collector",
+								Image:           r.images["plugin"],
+								ImagePullPolicy: imagePullPolicy,
+								Resources:       resources,
+								Command: []string{
+									"sh", "-c", `while true; do 
+								    	sleep 15m
+										logrotate --verbose /logrotate-config/csi
+									done`,
```

Comment:
Please explain why we need this elaborate command and not just the `logrotate` line.

Reply:
We need to run the container every 15 mins
And so we need a while loop running `logrotate command`

As it is container logroate cron job wont work so we need to explicitly run it

Reply:
First, there is a way to run cronjob in a container: https://github.com/Ekito/docker-cron
Second, even if you want to run manually, use `watch -n<time> <command>` instead of writing a bash loop